In [1]:
import pandas as pd



In [2]:
df = pd.read_csv(r"C:\Users\HP\Maths and statistics course codebasics\files\chapter4_assets\9_stock_returns_volatility\stock_returns.csv")
df

year  nvidia_returns  reliance_returns
0  2016          122.14             47.43
1  2017           83.91             12.09
2  2018          -50.03            -13.63
3  2019           82.19             22.42
4  2020          125.84             62.59
5  2021           50.53             33.70
6  2022           48.57             10.26

In [3]:
df.describe()

year  nvidia_returns  reliance_returns
count     7.000000        7.000000          7.000000
mean   2019.000000       66.164286         24.980000
std       2.160247       59.611670         25.412166
min    2016.000000      -50.030000        -13.630000
25%    2017.500000       49.550000         11.175000
50%    2019.000000       82.190000         22.420000
75%    2020.500000      103.025000         40.565000
max    2022.000000      125.840000         62.590000

In [5]:
df.nvidia_returns.std(),df.nvidia_returns.var()                   #get nvidia return std and variance

(59.61166995847453, 3553.551195238095)

In [6]:
df.reliance_returns.std(),df.reliance_returns.var()             

(25.41216637754444, 645.7782000000001)

Standard Deviation (std):

    NVIDIA Returns (std = 59.611):
        This value indicates that the returns for NVIDIA have a relatively high level of variability. A high standard deviation suggests that the returns fluctuate significantly around the mean (66.16).

    Reliance Returns (std = 25.421):
        This value is significantly lower than that of NVIDIA, indicating that Reliance returns are less volatile compared to NVIDIA returns. The returns of Reliance fluctuate less around the mean (24.98).

Variance:

    NVIDIA Returns (variance = 3553.551):
        Variance, which is the square of the standard deviation, also shows the extent of the variability. The higher the variance, the more spread out the returns are. For NVIDIA, a high variance reinforces the insight that its returns are highly volatile.

    Reliance Returns (variance = 645.772):
        This lower variance compared to NVIDIA indicates that Reliance's returns are more stable and less spread out.

Comparative Insights:

    Risk Assessment:
        NVIDIA's higher standard deviation and variance suggest that it is a riskier investment compared to Reliance. Investors may expect larger swings in returns, which could mean higher potential gains but also higher potential losses.
    Stability:
        Reliance, with its lower standard deviation and variance, appears to be a more stable investment. It might be preferable for risk-averse investors who prioritize stability over high returns.

Summary:

    NVIDIA:
        Mean Return: 66.16
        Std: 59.611
        Variance: 3553.551
        Insight: High returns but high risk/volatility.

    Reliance:
        Mean Return: 24.98
        Std: 25.421
        Variance: 645.772
        Insight: Moderate returns with lower risk/volatility.

These insights can help investors make informed decisions based on their risk tolerance and return expectations.